pkg/mqtrigger/scalermanager.go

```diff
@@ -0,0 +1,265 @@
+	resyncPeriod := 30 * time.Second
+	listWatch := k8sCache.NewListWatchFromClient(crdClient, "messagequeuetriggers", metav1.NamespaceAll, fields.Everything())
+	_, controller := k8sCache.NewInformer(listWatch, &fv1.MessageQueueTrigger{}, resyncPeriod, k8sCache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			go func() {
+				mqt := obj.(*fv1.MessageQueueTrigger)
+				logger.Debug("Create deployment for Scaler Object", zap.Any("mqt", mqt.ObjectMeta), zap.Any("maqt.Spec", mqt.Spec))
+				scaledObject := getScaledObject(mqt)
+				kedaClient, err := getKedaClient(mqt.ObjectMeta.Namespace)
+				if err != nil {
+					logger.Error("Failed to create KEDA client", zap.Error(err))
+				}
+				_, err = kedaClient.Create(scaledObject, metav1.CreateOptions{})
+				if err != nil {
+					logger.Error("Failed to create ScaledObject", zap.Error(err))
+				}
+				deployment := getDeploymentSpec(mqt, routerUrl)
+				_, err = deploymentsClient.Create(deployment)
+				if err != nil {
+					logger.Error("Failed to create deployment", zap.Error(err))
+				}
+			}()
+
+		},
+		UpdateFunc: func(_ interface{}, newObj interface{}) {
+			go func() {
+				mqt := newObj.(*fv1.MessageQueueTrigger)
+				logger.Debug("Update deployment for Scaler Object", zap.Any("mqt", mqt.ObjectMeta), zap.Any("maqt.Spec", mqt.Spec))
+				scaledObject := getScaledObject(mqt)
+				kedaClient, err := getKedaClient(mqt.ObjectMeta.Namespace)
+				if err != nil {
+					logger.Error("Failed to create KEDA client", zap.Error(err))
+				}
+				_, err = kedaClient.Update(scaledObject, metav1.UpdateOptions{})
+				if err != nil {
+					logger.Error("Failed to Update ScaledObject", zap.Error(err))
+				}
+			}()
+		},
+		DeleteFunc: func(obj interface{}) {
+			go func() {
+				mqt := obj.(*fv1.MessageQueueTrigger)
+				logger.Debug("Delete deployment for Scaler Object", zap.Any("mqt", mqt.ObjectMeta), zap.Any("maqt.Spec", mqt.Spec))
+				name := mqt.ObjectMeta.Name
+				kedaClient, err := getKedaClient(mqt.ObjectMeta.Namespace)
+				if err != nil {
+					logger.Error("Failed to create KEDA client", zap.Error(err))
+				}
+				err = kedaClient.Delete(name, &metav1.DeleteOptions{})
+				if err != nil {
+					logger.Error("Failed to Delete ScaledObject", zap.Error(err))
+				}
+				deletePolicy := metav1.DeletePropagationForeground
+				if err := deploymentsClient.Delete(mqt.ObjectMeta.Name, &metav1.DeleteOptions{
+					PropagationPolicy: &deletePolicy,
+				}); err != nil {
+					logger.Error("Failed to Delete Deployment", zap.Error(err))
+				}
+			}()
+		},
+	})
+	controller.Run(context.Background().Done())
+	return nil
+}
+
+func getScaledObject(mqt *fv1.MessageQueueTrigger) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "ScaledObject",
+			"apiVersion": "keda.k8s.io/v1alpha1",
+			"metadata": map[string]interface{}{
+				"name":      mqt.ObjectMeta.Name,
+				"namespace": mqt.ObjectMeta.Namespace,
```

Comment:
It would be good to add ownerReference to metadata, check https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/

Reply:
Done